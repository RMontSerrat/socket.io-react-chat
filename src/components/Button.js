import React from 'react';

const Button = ({ children, ...rest }) => {
  return (
    <button class="button" {...rest}>
      {children}
    </button>
  );
};

export default Button;
