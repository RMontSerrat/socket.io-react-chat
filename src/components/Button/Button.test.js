import React from "react";
import { render, cleanup } from "react-testing-library";
import Button from "./Button";

afterEach(cleanup);
describe('<Button />', () => {
  it("should render", () => {
    const { asFragment } = render(
      <Button>Click here</Button>
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it("should render a submit button", () => {
    const { asFragment } = render(
      <Button type="submit">Click here</Button>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});