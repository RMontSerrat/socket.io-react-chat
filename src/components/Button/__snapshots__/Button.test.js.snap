// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> should render 1`] = `
<DocumentFragment>
  .c0 {
  width: 100%;
  font-weight: bold;
  -webkit-transition: all 0.2s ease-in 0s;
  transition: all 0.2s ease-in 0s;
  border: 0;
  cursor: pointer;
  outline: none;
}

<button
    class="c0"
  >
    Click here
  </button>
</DocumentFragment>
`;

exports[`<Button /> should render a submit button 1`] = `
<DocumentFragment>
  .c0 {
  width: 100%;
  font-weight: bold;
  -webkit-transition: all 0.2s ease-in 0s;
  transition: all 0.2s ease-in 0s;
  border: 0;
  cursor: pointer;
  outline: none;
}

<button
    class="c0"
    type="submit"
  >
    Click here
  </button>
</DocumentFragment>
`;
